SwiftGit2/Credentials.swift

```diff
@@ -46,7 +61,7 @@ internal func credentialsCallback(
 	// Find username_from_url
 	let name = username.map(String.init(cString:))
 
-	switch Credentials.fromPointer(payload!) {
+	switch Credentials.fromPointer(payload!) ?? .default {
```

Comment:
Why is the coalescing operator needed here? Instead we could remove it and set it int the default case
```swift
default:
result = git_cred_default_new(cred)
```